skimage/registration/_affine.py

```diff
@@ -0,0 +1,584 @@
+    weights: ndarray
+        Weights or mask
+    channel_axis: int | None
+        Index of the channel axis
+    matrix: ndarray
+        Initial value of the transform, here matrix are parameters
+    model: str
+        Motion model: "affine", "translation" or "euclidean"
+    method: str
+        Minimization method for scipy.optimization.minimize
+    options: dict
+        options for scipy.optimization.minimize
+    cost: function
+        Cost function minimize
+    vector_to_matrix: function(param, ndim) -> ndarray
+        Convert a vector of parameters to a matrix
+
+    Returns
+    -------
+    matrix:
+        Homogeneous transform matrix
+
+    Raises
+    ------
+    NotImplementedError for unsupported motion models.
+
+    Reference
+    ---------
+    .. [1] Studholme C, Hill DL, Hawkes DJ. Automated 3-D registration of MR
+        and CT images of the head. Med Image Anal. 1996 Jun;1(2):163-75.
+    .. [2] J. Nunez-Iglesias, S. van der Walt, and H. Dashnow, Elegant SciPy:
+        The Art of Scientific Python. O’Reilly Media, Inc., 2017.
+
+    """
+
+    if channel_axis is not None:
+        reference_image = np.moveaxis(reference_image, channel_axis, 0)
+        moving_image = np.moveaxis(moving_image, channel_axis, 0)
+    else:
+        reference_image = np.expand_dims(reference_image, 0)
+        moving_image = np.expand_dims(moving_image, 0)
+
+    ndim = reference_image.ndim - 1
+
+    if matrix is None:
+        if model.lower() == "translation":
+            matrix = np.zeros(ndim)
+        elif model.lower() == "euclidean":
+            nrotations = len([x for x in combinations(range(ndim), 2)])
+            matrix = np.zeros(ndim + nrotations)
+        elif model.lower() == "affine":
+            matrix = np.zeros(ndim * (ndim + 1))
+        else:
+            raise NotImplementedError(f"Motion model {model} not implemented.")
+
+    cost = partial(
+        _param_cost,
+        reference_image=reference_image,
+        moving_image=moving_image,
+        weights=weights,
+        cost=cost,
+        model=model,
+    )
+
+    result = minimize(cost, x0=matrix, method=method, options=options)
+
+    return result.x
+
+
+def affine(
+    reference_image,
+    moving_image,
+    *,
+    weights=None,
+    channel_axis=None,
+    matrix=None,
+    model="affine",
+    solver=lucas_kanade_affine_solver,
+    pyramid_downscale=2,
+    pyramid_minimum_size=32,
+):
+    """
+    Coarse-to-fine affine motion estimation between two images
+
+    Parameters
+    ----------
+    reference_image : ndarray
+        The first gray scale image of the sequence.
+    moving_image : ndarray
+        The second gray scale image of the sequence.
+    weights : ndarray | None
+        The weights array with same shape as reference_image
+    channel_axis: int | None
+        Index of the channel axis
+    matrix : ndarray | None
+        Intial guess for the homogeneous transformation matrix
+    model: str
+        Motion model
```

Comment:
The docstrings for each solver specify that we can use the model options "affine", "translation" or "euclidean", but most users would interact with this option at the level of this function so we should make that especially clear in this docstring.

As a lower bound, give them the options: `Motion model: "affine", "translation" or "euclidean".` 

In addition, it might be wise to spend a couple sentences expanding on the difference/constraints of a model limited to translation or euclidean registration alone versus affine registration so a user with less experience could choose without searching elsewhere.

Reply:
I added the following to the affine documentation:
```  
  model: str
        Motion model: "translation", "euclidean" or "affine" where euclidean motion
        corresponds to rigid motion (rotation and translation) and affine motion can
        represent change in scale, shear, rotation and translations.
```

Reply:
Each function with motion model now has now the doc string
```
Motion model: "affine", "translation" or "euclidean"
```